src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,476 @@
+          );
+        }
 
+        // Return null if conditions are not met
+        return null;
+      };
+
+      const itemsNotFound = () => {
+        if (itemsList.length === 0) {
+          return (
+            <Box
+              key="items-not-found-text-id"
+              id="items-not-found-text-id"
+              marginTop="xs"
+            >
+              No options found
+            </Box>
+          );
+        }
+        return null; // Return null when itemsList.length !== 0
+      };
+
+      /**
+       * Toggle between viewing all items and a limited number of default items.
+       * Updates the state to control the visibility and height of the items list.
+       */
+      const viewAllItems = () => {
+        setViewAllLabel((prevProp) =>
+          prevProp === "View all" ? "View less" : "View all"
+        );
+        setListHeight((prevProp) =>
+          prevProp === MAXIMUM_ITEMS_LIST_HEIGHT
+            ? DEFAULT_ITEMS_LIST_HEIGHT
+            : MAXIMUM_ITEMS_LIST_HEIGHT
+        );
+        setListItemsCount((prevProp) =>
+          prevProp === defaultItemsVisible ? items.length : defaultItemsVisible
+        );
+
+        if (listItemsCount === defaultItemsVisible) {
+          setItemsList(items);
+        } else {
+          displayDefaultItems();
+        }
+      };
+
+      const buttonTextLabel = () => {
         return (
-          <MultiSelectDialog
-            {...commonProps}
-            onChange={dialogOnChange}
-            onMixedStateChange={onMixedStateChange}
-            onApply={onApply}
-            ref={ref}
-            {...rest}
-          />
+          <Box
+            __css={
+              getSelectedItemsCount > 0
+                ? styles.buttonTextLabel
+                : null /* or pass an empty object {} if you don't want any styles */
+            }
+            title={buttonText}
+          >
+            {buttonText}
+          </Box>
         );
-      }
+      };
 
-      return null;
+      // const itemsTotal = selectedItems[id].items.length;
+      const itemPlural = getSelectedItemsCount === 1 ? "" : "s";
+      const ariaLabelValue = `${buttonText}, ${getSelectedItemsCount} item${itemPlural} currently selected`;
+      return (
+        <Box id={id} __css={styles} {...rest}>
+          <Accordion
+            accordionData={[
+              {
+                accordionType: "default",
+                // Pass the ref for interaction with the accordion button.
+                buttonInteractionRef: accordianButtonRef,
+                label: buttonTextLabel(),
+                panel: displayAccordionData(),
+              },
+            ]}
+            ariaLabel={ariaLabelValue}
+            id="multi-select-accordion-id"
+            isDefaultOpen={isDefaultOpen}
+            isAlwaysRendered
+            panelMaxHeight={listHeight}
+            sx={{
+              ...styles.accordionStyles,
+            }}
+          />
+          <MultiSelectItemsCountButton
```

Comment:
Render this button conditionally `{selectedItemsCount > 0 && <MultiSelectItemsCountButton/>}`